File: 1966-frequency-of-the-most-frequent-element/frequency-of-the-most-frequent-element.ts
function maxFrequency(nums: number[], k: number): number {
    nums.sort((a,b)=> (a-b));

    let result : number = 1;
    let start : number = 0;
    let sum : number = 0;
    let n : number = nums.length;

    for(let end = 0; end < n; end++) {
        sum += nums[end];

        while(nums[end] * (end - start + 1)  - sum > k) {
            sum -= nums[start++];
        }

        result = Math.max(result,end-start+1);
    }


    return result;
};

File: 1966-frequency-of-the-most-frequent-element/frequency-of-the-most-frequent-element.java
class Solution {
    public int maxFrequency(int[] nums, int k) {
    Arrays.sort(nums);

    int result = 1;
    int start = 0;
    long sum = 0;
    int n = nums.length;

    for(int end = 0; end < n; end++) {
        sum += nums[end];

        while((long) nums[end] * (end - start + 1)  - sum > k) {
            sum -= nums[start++];
        }

        result = Math.max(result,end-start+1);
    }


    return result;
    }
}